app/controllers/samples_controller.rb

```diff
@@ -164,8 +170,38 @@ def bulk_destroy
     redirect_to samples_path, notice: 'Samples were successfully deleted.'
   end
 
+  def transfer
+    new_owner = Institution.find_by(uuid: params["institution_id"])
+    if new_owner.nil?
+      message = "Destination Institution does not exists"
+    else
+      not_transferred = []
+      change_ownership(new_owner, not_transferred)
+      message = (not_transferred.empty?) ? "All samples has been transferred successfully." : "#{not_transferred.length} #{"sample".pluralize(not_transferred.length)} failed on the transfer."
+    end
+
+    flash[:notice] = message
+    render json: {status: :ok}
+  end
+
   private
 
+  def change_ownership(new_owner, not_transferred)
```

Comment:
There is a **security issue** is this method: we don't check that the current user has access to the samples. I can easily forge a request that will transfer any sample from any institution or site!